arrow-libs/core/arrow-core-data/src/test/kotlin/arrow/core/NonEmptyListTest.kt

```diff
@@ -90,5 +90,137 @@ class NonEmptyListTest : UnitSpec() {
         }
       }
     }
+
+    "zip2" {
+      forAll(Gen.nonEmptyList(Gen.int()), Gen.nonEmptyList(Gen.int())) { a, b ->
+        val result = a.zip(b)
+        val expected = a.all.zip(b.all).let(NonEmptyList.Companion::fromListUnsafe)
```

Comment:
aren't we testing against our own zip version here?

Reply:
Yes, but that `zip` is tested against Kotlin Std `zip` in `IterableTest` so it assumes our `zip` is correct/tested. Which is added in this PR also.